base_substate/tests/test_base_substate.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2019 Akretion Mourad EL HADJ MIMOUNE
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests import common
+from .common import setup_test_model
+from .sale_test import SaleTest, LineTest
+
+
+@common.at_install(False)
+@common.post_install(True)
+class TestBaseSubstate(common.SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestBaseSubstate, cls).setUpClass()
+        # from .sale_test import SaleTest, LineTest
+
+        setup_test_model(cls.env, [SaleTest, LineTest])
```

Comment:
you should handle `tearDownClass` as well.
Please have a look here 
https://github.com/OCA/search-engine/blob/12.0/connector_search_engine/tests/test_all.py#L15
https://github.com/OCA/search-engine/blob/12.0/connector_search_engine/tests/models_mixin.py
https://github.com/OCA/search-engine/blob/12.0/connector_search_engine/tests/models.py
(maybe it's time to move this to a lib :D )

Reply:
I add new method in models_mixin.py to setup several fake models at the same time to avoid bug of many2one field that reference fake model.